state-chain/pallets/cf-elections/src/lib.rs

```diff
@@ -739,19 +740,37 @@ pub mod pallet {
 								},
 								validator_id,
 							)
-						).filter(|(_, validator_id)| {
-							ContributingAuthorities::<T, I>::contains_key(validator_id)
-						}).filter_map(|(vote_components, validator_id)| {
-							<<T::ElectoralSystem as ElectoralSystem>::Vote as VoteStorage>::components_into_authority_vote(vote_components, |shared_data_hash| {
-								// We don't bother to check if the reference has expired, as if we have the data we may as well use it, even if it was provided after the shared data reference expired (But before the reference was cleaned up `on_finalize`).
-								Ok(SharedData::<T, I>::get(shared_data_hash))
-							}).map(|vote| vote.map(|vote| (vote.0, vote.1, validator_id))).transpose()
-						}).filter_map_ok(|(properties, authority_vote, validator_id)| match authority_vote {
-							AuthorityVote::Vote(vote) => Some((properties, vote, validator_id)),
-							_ => None,
+						).map(|(vote, validator_id)| {
+							(if ContributingAuthorities::<T, I>::contains_key(&validator_id) {
+								Some(vote)
+							} else {
+								None
+							}, validator_id)
+						}).map(|(vote_components, validator_id)| {
```

Comment:
Why not merge these two maps? The if/else conditions are basically the same.

Reply:
No strong reason, just to separate the two ideas clearly

Reply:
I changed this, see what you meant on second look